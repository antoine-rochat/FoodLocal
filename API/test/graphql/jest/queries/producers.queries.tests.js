const { graphql } = require('graphql');
const { makeExecutableSchema } = require('graphql-tools');
const producersService = require('../../../../src/graphql/services/producers.services');
const { resolvers, schema: typeDefs } = require('../../../../src/graphql/graphqlConfig');
const clearDB = require('../../clearDB');const { Products: ProductModel, ProductType: ProductTypeModel, ProductTypeCategory: ProductTypeCategoryModel } = require(
  '../../../../src/graphql/models/products.modelgql');
const { queryObjAllProducers, queryObjProducerWithCorrectId, queryObjProducerWithWrongId } = require('./Objects/QueryObjsProducers');

const schema = makeExecutableSchema({ typeDefs, resolvers });

let benoit = {};
let antoine = {};
let pomme = {};
let poire = {};

const beforeEachFunc = () => async() => {
  // ---------------------------------------- on supprime tout le contenu de la DB ----------------------------------------
  await clearDB();

  // ------------------------------------------- on ajoute le contenu de départ -------------------------------------------
  // Création du type Fruit
  let productTypeCategory = {
    name: 'Fruits',
    image: 'ceci est une image de fruits encodée en base64!'
  };

  // on ajoute le productTypeCategory
  productTypeCategory = (await new ProductTypeCategoryModel(productTypeCategory).save()).toObject();

  // on ajoute 2 productType
  const productTypePomme = {
    name: 'Pomme',
    image: 'ceci est une image de pomme encodée en base64!',
    categoryId: productTypeCategory.id,
    producersIds: []
  };
  pomme = (await new ProductTypeModel(productTypePomme).save()).toObject();

  const productTypePoire = {
    name: 'Poire',
    image: 'ceci est une image de poire encodée en base64!',
    categoryId: productTypeCategory.id,
    producersIds: []
  };
  poire = (await ProductTypeModel(productTypePoire)
    .save()).toObject();

  // on ajoute le contenu de départ
  // on ajoute 2 producteur avec leur point de vente

  benoit = {
    firstname: 'Benoît',
    lastname: 'Schöpfli',
    email: 'benoit@paysan.ch',
    password: '1234abcd',
    image: 'Ceci est une image encodée en base64!',
    phoneNumber: '0761435196',
    description: 'Un chouet gaillard!',
    website: 'benoit-paysan.ch',
    salespoint: {
      name: 'Chez moi',
      address: {
        number: 6,
        street: 'Chemin de par ici',
        city: 'Yverdon',
        postalCode: '1400',
        state: 'Vaud',
        country: 'Suisse',
        longitude: 1.1234567,
        latitude: 1.123456789
      },
      schedule:
        {
          monday: [
            {
              openingHour: '08:00',
              closingHour: '12:00'
            },
            {
              openingHour: '13:00',
              closingHour: '18:00'
            }
          ],
          tuesday: [],
          wednesday: [
            {
              openingHour: '08:00',
              closingHour: '12:00'
            }
          ],
          thursday: [],
          friday: [
            {
              openingHour: '08:00',
              closingHour: '12:00'
            }
          ],
          saturday: [],
          sunday: []
        }
    },
    products: [
      {
        description: 'Une pomme moins bonne que celle d\'antoine!',
        productTypeId: pomme.id
      },
      {
        description: 'Une poire bof bof!',
        productTypeId: poire.id
      }
    ]
  };

  antoine = {
    firstname: 'Antoine',
    lastname: 'Rochaille',
    email: 'antoine@paysan.ch',
    password: '1234abcd',
    image: 'Ceci est l\'image d\'un tueur encodée en base64!',
    phoneNumber: '0761435196',
    description: 'Un vrai payouz!',
    salespoint: {
      name: 'Chez lui, perdu au milieu de rien',
      address: {
        number: 12,
        street: 'Chemin de par ici',
        city: 'Quelque part par là-bas',
        postalCode: '1234',
        state: 'Vaud',
        country: 'Suisse',
        longitude: 1.1234796,
        latitude: 1.123418029
      },
    },
    products: [
      {
        description: 'Une pomme meilleure que celle de Benoît!',
        productTypeId: pomme.id
      }
    ]
  };
  benoit = (await producersService.addProducer(benoit)).toObject();

  antoine = (await producersService.addProducer(antoine)).toObject();
};

describe('Testing graphql request producers', () => {
  // -------------------------Producers()------------------------------------- //
  describe('Testing producers()', () => {
    beforeEach(beforeEachFunc());
    it('Getting all producers', async (done) => {
      const { query, variables, context } = queryObjAllProducers;
      const result = await graphql(schema, query, null, context, variables);
      expect.assertions(1);
      expect(result).toMatchSnapshot();
      done();
    });
  });
  // ----------------------Producer(id)-------------------------------------- //
  describe('Testing producer(id)', () => {
    beforeEach(beforeEachFunc());
    it('Getting producer by id', async (done) => {
      const { query, context } = queryObjProducerWithCorrectId;
      const variables = { id: antoine.id };
      const result = await graphql(schema, query, null, context, variables);
      expect.assertions(1);
      expect(result).toMatchSnapshot();
      done();
    });
    it('Fail getting producer with wrong id', async (done) => {
      const { query, context } = queryObjProducerWithWrongId;
      const variables = { id: 'abcdefabcdefabcdefabcdef' };
      const result = await graphql(schema, query, null, context, variables);
      expect.assertions(1);
      expect(result.data.producer).toBeNull();
      done();
    });
  });


});
