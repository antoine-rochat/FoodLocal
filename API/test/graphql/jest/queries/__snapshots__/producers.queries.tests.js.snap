// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding empty producer 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Variable "$producer" got invalid value {}; Field value.firstname of required type String! was not provided.],
    [GraphQLError: Variable "$producer" got invalid value {}; Field value.lastname of required type String! was not provided.],
    [GraphQLError: Variable "$producer" got invalid value {}; Field value.email of required type String! was not provided.],
    [GraphQLError: Variable "$producer" got invalid value {}; Field value.password of required type String! was not provided.],
  ],
}
`;

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding producer correctly 1`] = `
Object {
  "data": Object {
    "addProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding producer with empty email 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: producers validation failed: email: Path \`email\` is required.],
  ],
}
`;

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding producer with empty firstname 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: producers validation failed: firstname: Path \`firstname\` is required.],
  ],
}
`;

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding producer with empty lastname 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: producers validation failed: lastname: Path \`lastname\` is required.],
  ],
}
`;

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding producer with empty password 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: producers validation failed: password: Path \`password\` is required.],
  ],
}
`;

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding producer with null email 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Variable "$producer" got invalid value { firstname: "Antoine", lastname: "Rochaille", email: null, password: "1234abcd", image: "Ceci est l'image d'un tueur encodée en base64!", phoneNumber: "0761435196", description: "Un vrai payouz!" }; Expected non-nullable type String! not to be null at value.email.],
  ],
}
`;

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding producer with null firstname 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Variable "$producer" got invalid value { firstname: null, lastname: "Rochaille", email: "antoine@paysan.ch", password: "1234abcd", image: "Ceci est l'image d'un tueur encodée en base64!", phoneNumber: "0761435196", description: "Un vrai payouz!" }; Expected non-nullable type String! not to be null at value.firstname.],
  ],
}
`;

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding producer with null lastname 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Variable "$producer" got invalid value { firstname: "Antoine", lastname: null, email: "antoine@paysan.ch", password: "1234abcd", image: "Ceci est l'image d'un tueur encodée en base64!", phoneNumber: "0761435196", description: "Un vrai payouz!" }; Expected non-nullable type String! not to be null at value.lastname.],
  ],
}
`;

exports[`Testing graphql request producers MUTATION producers Testing addProducer Adding producer with null password 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Variable "$producer" got invalid value { firstname: "Antoine", lastname: "Rochaille", email: "antoine@paysan.ch", password: null, image: "Ceci est l'image d'un tueur encodée en base64!", phoneNumber: "0761435196", description: "Un vrai payouz!" }; Expected non-nullable type String! not to be null at value.password.],
  ],
}
`;

exports[`Testing graphql request producers MUTATION producers Testing validateAProducer (Producer id) Changing validation producer to false 1`] = `
Object {
  "data": Object {
    "validateAProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers Testing validateAProducer (Producer id) Changing validation producer to true 1`] = `
Object {
  "data": Object {
    "validateAProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": true,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers deleteProducer Delete correctly producer 1`] = `
Object {
  "data": Object {
    "deleteProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers deleteProducer Delete unknow user (wrong id) 1`] = `
Object {
  "data": Object {
    "deleteProducer": null,
  },
}
`;

exports[`Testing graphql request producers MUTATION producers updateProducer updateProducer description correctly 1`] = `
Object {
  "data": Object {
    "updateProducer": Object {
      "description": "testDesc",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers updateProducer updateProducer email correctly 1`] = `
Object {
  "data": Object {
    "updateProducer": Object {
      "description": "Un vrai payouz!",
      "email": "test@test.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers updateProducer updateProducer email correctly 2`] = `
Object {
  "data": Object {
    "updateProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "test",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers updateProducer updateProducer firstname correctly 1`] = `
Object {
  "data": Object {
    "updateProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "bob",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers updateProducer updateProducer firstname correctly without giving all object 1`] = `
Object {
  "data": Object {
    "updateProducer": Object {
      "description": null,
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "bob",
      "followers": Array [],
      "followingProducers": Array [],
      "image": null,
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": null,
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers updateProducer updateProducer image correctly 1`] = `
Object {
  "data": Object {
    "updateProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "test",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers updateProducer updateProducer lastname correctly 1`] = `
Object {
  "data": Object {
    "updateProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "test",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers updateProducer updateProducer phoneNumber correctly 1`] = `
Object {
  "data": Object {
    "updateProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "7357",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers MUTATION producers updateProducer updateProducer website correctly 1`] = `
Object {
  "data": Object {
    "updateProducer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": "www.test.ch",
    },
  },
}
`;

exports[`Testing graphql request producers QUERY producers Testing filterProducers(ProductType) Getting producers selling apple 1`] = `
Object {
  "data": Object {
    "filterProducers": Array [
      Object {
        "description": "Un chouet gaillard!",
        "email": "benoit@paysan.ch",
        "emailValidated": false,
        "firstname": "Benoît",
        "image": "Ceci est une image encodée en base64!",
        "isValidated": false,
        "lastname": "Schöpfli",
        "password": "1234abcd",
        "phoneNumber": "0761435196",
        "products": Array [
          Object {
            "description": "Une pomme monnnnstre bonne!",
            "productType": Object {
              "category": Object {
                "image": "ceci est une image de fruits encodée en base64!",
                "name": "Fruits",
              },
              "image": "ceci est une image de pomme encodée en base64!",
              "name": "Pomme",
            },
          },
          Object {
            "description": "Une poire de folie!",
            "productType": Object {
              "category": Object {
                "image": "ceci est une image de fruits encodée en base64!",
                "name": "Fruits",
              },
              "image": "ceci est une image de poire encodée en base64!",
              "name": "Poire",
            },
          },
        ],
        "salespoint": Object {
          "address": Object {
            "city": "Yverdon",
            "country": "Suisse",
            "latitude": 1.123456789,
            "longitude": 1.1234567,
            "number": 6,
            "postalCode": "1400",
            "street": "Chemin de par ici",
          },
          "name": "Chez moi",
          "schedule": Object {
            "friday": Array [
              Object {
                "closingHour": "12:00",
                "openingHour": "08:00",
              },
            ],
            "monday": Array [
              Object {
                "closingHour": "12:00",
                "openingHour": "08:00",
              },
              Object {
                "closingHour": "18:00",
                "openingHour": "13:00",
              },
            ],
            "saturday": Array [],
            "sunday": Array [],
            "thursday": Array [],
            "tuesday": Array [],
            "wednesday": Array [
              Object {
                "closingHour": "12:00",
                "openingHour": "08:00",
              },
            ],
          },
        },
        "website": "benoitpaysan.ch",
      },
      Object {
        "description": "Un vrai payouz!",
        "email": "antoine@paysan.ch",
        "emailValidated": false,
        "firstname": "Antoine",
        "image": "Ceci est l'image d'un tueur encodée en base64!",
        "isValidated": false,
        "lastname": "Rochaille",
        "password": "1234abcd",
        "phoneNumber": "0761435196",
        "products": Array [
          Object {
            "description": "Une pomme monnnnstre bonne!",
            "productType": Object {
              "category": Object {
                "image": "ceci est une image de fruits encodée en base64!",
                "name": "Fruits",
              },
              "image": "ceci est une image de pomme encodée en base64!",
              "name": "Pomme",
            },
          },
        ],
        "salespoint": null,
        "website": null,
      },
    ],
  },
}
`;

exports[`Testing graphql request producers QUERY producers Testing filterProducers(ProductType) Getting producers selling perry 1`] = `
Object {
  "data": Object {
    "filterProducers": Array [
      Object {
        "description": "Un chouet gaillard!",
        "email": "benoit@paysan.ch",
        "emailValidated": false,
        "firstname": "Benoît",
        "image": "Ceci est une image encodée en base64!",
        "isValidated": false,
        "lastname": "Schöpfli",
        "password": "1234abcd",
        "phoneNumber": "0761435196",
        "products": Array [
          Object {
            "description": "Une pomme monnnnstre bonne!",
            "productType": Object {
              "category": Object {
                "image": "ceci est une image de fruits encodée en base64!",
                "name": "Fruits",
              },
              "image": "ceci est une image de pomme encodée en base64!",
              "name": "Pomme",
            },
          },
          Object {
            "description": "Une poire de folie!",
            "productType": Object {
              "category": Object {
                "image": "ceci est une image de fruits encodée en base64!",
                "name": "Fruits",
              },
              "image": "ceci est une image de poire encodée en base64!",
              "name": "Poire",
            },
          },
        ],
        "salespoint": Object {
          "address": Object {
            "city": "Yverdon",
            "country": "Suisse",
            "latitude": 1.123456789,
            "longitude": 1.1234567,
            "number": 6,
            "postalCode": "1400",
            "street": "Chemin de par ici",
          },
          "name": "Chez moi",
          "schedule": Object {
            "friday": Array [
              Object {
                "closingHour": "12:00",
                "openingHour": "08:00",
              },
            ],
            "monday": Array [
              Object {
                "closingHour": "12:00",
                "openingHour": "08:00",
              },
              Object {
                "closingHour": "18:00",
                "openingHour": "13:00",
              },
            ],
            "saturday": Array [],
            "sunday": Array [],
            "thursday": Array [],
            "tuesday": Array [],
            "wednesday": Array [
              Object {
                "closingHour": "12:00",
                "openingHour": "08:00",
              },
            ],
          },
        },
        "website": "benoitpaysan.ch",
      },
    ],
  },
}
`;

exports[`Testing graphql request producers QUERY producers Testing producer(ProducerId) should fail getting a producer beaucse invalid id received 1`] = `
Object {
  "data": Object {
    "producer": null,
  },
  "errors": Array [
    [GraphQLError: Received producer.id is invalid!],
  ],
}
`;

exports[`Testing graphql request producers QUERY producers Testing producer(ProducerId) should fail getting a producer because unknown id received 1`] = `
Object {
  "data": Object {
    "producer": null,
  },
}
`;

exports[`Testing graphql request producers QUERY producers Testing producer(ProducerId) should get a producer by id (with schedule) 1`] = `
Object {
  "data": Object {
    "producer": Object {
      "description": "Un chouet gaillard!",
      "email": "benoit@paysan.ch",
      "emailValidated": false,
      "firstname": "Benoît",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est une image encodée en base64!",
      "isValidated": false,
      "lastname": "Schöpfli",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
        Object {
          "description": "Une poire de folie!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de poire encodée en base64!",
            "name": "Poire",
          },
        },
      ],
      "salespoint": Object {
        "address": Object {
          "city": "Yverdon",
          "country": "Suisse",
          "latitude": 1.123456789,
          "longitude": 1.1234567,
          "number": 6,
          "postalCode": "1400",
          "street": "Chemin de par ici",
        },
        "name": "Chez moi",
        "schedule": Object {
          "friday": Array [
            Object {
              "closingHour": "12:00",
              "openingHour": "08:00",
            },
          ],
          "monday": Array [
            Object {
              "closingHour": "12:00",
              "openingHour": "08:00",
            },
            Object {
              "closingHour": "18:00",
              "openingHour": "13:00",
            },
          ],
          "saturday": Array [],
          "sunday": Array [],
          "thursday": Array [],
          "tuesday": Array [],
          "wednesday": Array [
            Object {
              "closingHour": "12:00",
              "openingHour": "08:00",
            },
          ],
        },
      },
      "website": "benoitpaysan.ch",
    },
  },
}
`;

exports[`Testing graphql request producers QUERY producers Testing producer(ProducerId) should get a producer by id (without schedule) 1`] = `
Object {
  "data": Object {
    "producer": Object {
      "description": "Un vrai payouz!",
      "email": "antoine@paysan.ch",
      "emailValidated": false,
      "firstname": "Antoine",
      "followers": Array [],
      "followingProducers": Array [],
      "image": "Ceci est l'image d'un tueur encodée en base64!",
      "isValidated": false,
      "lastname": "Rochaille",
      "phoneNumber": "0761435196",
      "products": Array [
        Object {
          "description": "Une pomme monnnnstre bonne!",
          "productType": Object {
            "category": Object {
              "image": "ceci est une image de fruits encodée en base64!",
              "name": "Fruits",
            },
            "image": "ceci est une image de pomme encodée en base64!",
            "name": "Pomme",
          },
        },
      ],
      "salespoint": null,
      "website": null,
    },
  },
}
`;

exports[`Testing graphql request producers QUERY producers Testing producerWaitingForValidation() should get all producers waiting for validation 1`] = `
Object {
  "data": Object {
    "producersWaitingForValidation": Array [
      Object {
        "description": "Un chouet gaillard!",
        "email": "benoit@paysan.ch",
        "emailValidated": false,
        "firstname": "Benoît",
        "image": "Ceci est une image encodée en base64!",
        "isValidated": false,
        "lastname": "Schöpfli",
        "password": "1234abcd",
        "phoneNumber": "0761435196",
        "products": Array [
          Object {
            "description": "Une pomme monnnnstre bonne!",
            "productType": Object {
              "category": Object {
                "image": "ceci est une image de fruits encodée en base64!",
                "name": "Fruits",
              },
              "image": "ceci est une image de pomme encodée en base64!",
              "name": "Pomme",
            },
          },
          Object {
            "description": "Une poire de folie!",
            "productType": Object {
              "category": Object {
                "image": "ceci est une image de fruits encodée en base64!",
                "name": "Fruits",
              },
              "image": "ceci est une image de poire encodée en base64!",
              "name": "Poire",
            },
          },
        ],
        "salespoint": Object {
          "address": Object {
            "city": "Yverdon",
            "country": "Suisse",
            "latitude": 1.123456789,
            "longitude": 1.1234567,
            "number": 6,
            "postalCode": "1400",
            "street": "Chemin de par ici",
          },
          "name": "Chez moi",
          "schedule": Object {
            "friday": Array [
              Object {
                "closingHour": "12:00",
                "openingHour": "08:00",
              },
            ],
            "monday": Array [
              Object {
                "closingHour": "12:00",
                "openingHour": "08:00",
              },
              Object {
                "closingHour": "18:00",
                "openingHour": "13:00",
              },
            ],
            "saturday": Array [],
            "sunday": Array [],
            "thursday": Array [],
            "tuesday": Array [],
            "wednesday": Array [
              Object {
                "closingHour": "12:00",
                "openingHour": "08:00",
              },
            ],
          },
        },
        "website": "benoitpaysan.ch",
      },
    ],
  },
}
`;

exports[`Testing graphql request producers QUERY producers Testing producers() should get all producers 1`] = `
Object {
  "errors": Array [
    [GraphQLError: Syntax Error: Expected Name, found <EOF>],
  ],
}
`;
